plugins/modules/ec2_vpc_endpoint.py

```diff
@@ -247,9 +254,9 @@ def setup_creation(client, module):
 
     if not endpoint_id:
         # Try to use the module parameters to match any existing endpoints
-        all_endpoints = get_endpoints(client, module, endpoint_id)
-        if len(all_endpoints["VpcEndpoints"]) > 0:
-            for endpoint in all_endpoints["VpcEndpoints"]:
+        all_endpoints = get_endpoints(client, module.params, endpoint_id)
+        if len(all_endpoints) > 0:
```

Comment:
I don't think this `if len(all_endpoints) > 0` is necessary, if `all_endpoints` is an empty list the for loop just won't run.